### PR TITLE
COMMERCE-178 Check permissions into jsps and soy

### PR DESCRIPTION
**Pull request by alessiorendina**
This is the first part of the PR, I will add more permission checks. Do NOT merge it yet.

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test

File: commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/servlet/taglib/ui/CPDefinitionOptionsScreenNavigationEntry.java [hunk 1]
```diff
@@ -83,7 +90,21 @@ public boolean isVisible(User user, CPDefinition cpDefinition) {
 			return false;
 		}
 
-		return true;
+		PermissionChecker permissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		boolean hasViewCPDefinitionPermission = false;
```
Comment (ethanbustad) [inline commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/servlet/taglib/ui/CPDefinitionOptionsScreenNavigationEntry.java]: This variable name is a bit problematic... generally `is*` and `has*` names are reserved for methods.

But, wouldn't this be simpler to read anyway if we just got rid of the variable? It should work the same to just `return _cpDefinitionModelResourcePermission.contains...` in the `try` block, and `return false` at the end of the `catch` block. That seems more natural to me.

**Comment by alessiorendina** [comment]
ci:retest

**Comment by alessiorendina** [comment]
ci:retest

**Comment by ethanbustad** [comment]
ci:retest

**Comment by ethanbustad** [comment]
@Commando18 I think this most recent failure is an accurate one. The error message above the stacktrace is this:
```
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-definitions-web/build/jspc/org/apache/jsp/view_005finstances_jsp.java:407: error: cannot find symbol
     [echo]       _jspx_th_c_if_0.setTest( cpDefinitionsDisplayContext.hasPermission(CPActionKeys.MANAGE_CATALOG) );
     [echo]                                ^
     [echo]   symbol:   variable cpDefinitionsDisplayContext
     [echo]   location: class view_005finstances_jsp
```

Looks like there's a compile problem in `view_instances.jsp`. Can you take a look?

**Comment by alessiorendina** [comment]
ci:retest

**Comment by alessiorendina** [comment]
ci:stop

**Comment by alessiorendina** [comment]
ci:retest

**Comment by ethanbustad** [comment]
Another JSP compile error. `CPInstanceDisplayContext` doesn't have a `hasPermission` method.

**Comment by alessiorendina** [comment]
ci:retest

**Comment by marco-leo** [comment]
ci:retest

**Comment by pyoo47** [comment]
ci:test

**Comment by pyoo47** [comment]
Here are the actual compile failures that are coming up.

```
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/servlet/taglib/ui/CPDefinitionInstancesScreenNavigationEntry.java:122: error: constructor CPInstanceDisplayContext in class CPInstanceDisplayContext cannot be applied to given types;
     [echo] 				new CPInstanceDisplayContext(
     [echo] 				^
     [echo]   required: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper,PortletResourcePermission
     [echo]   found: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper
     [echo]   reason: actual and formal argument lists differ in length
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/servlet/taglib/ui/CPInstanceDetailsScreenNavigationEntry.java:118: error: constructor CPInstanceDisplayContext in class CPInstanceDisplayContext cannot be applied to given types;
     [echo] 				new CPInstanceDisplayContext(
     [echo] 				^
     [echo]   required: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper,PortletResourcePermission
     [echo]   found: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper
     [echo]   reason: actual and formal argument lists differ in length
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/EditCPInstanceMVCRenderCommand.java:65: error: constructor CPInstanceDisplayContext in class CPInstanceDisplayContext cannot be applied to given types;
     [echo] 				new CPInstanceDisplayContext(
     [echo] 				^
     [echo]   required: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper,PortletResourcePermission
     [echo]   found: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper
     [echo]   reason: actual and formal argument lists differ in length
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/portlet/action/ViewCPInstancesMVCRenderCommand.java:64: error: constructor CPInstanceDisplayContext in class CPInstanceDisplayContext cannot be applied to given types;
     [echo] 				new CPInstanceDisplayContext(
     [echo] 				^
     [echo]   required: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper,PortletResourcePermission
     [echo]   found: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper
     [echo]   reason: actual and formal argument lists differ in length
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/display/context/CPInstancePricingInfoDisplayContext.java:52: error: constructor CPInstanceDisplayContext in class CPInstanceDisplayContext cannot be applied to given types;
     [echo] 		super(
     [echo] 		^
     [echo]   required: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper,PortletResourcePermission
     [echo]   found: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper
     [echo]   reason: actual and formal argument lists differ in length
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/display/context/CPInstanceShippingInfoDisplayContext.java:50: error: constructor CPInstanceDisplayContext in class CPInstanceDisplayContext cannot be applied to given types;
     [echo] 		super(
     [echo] 		^
     [echo]   required: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper,PortletResourcePermission
     [echo]   found: ActionHelper,HttpServletRequest,CommercePriceFormatter,ModelResourcePermission<CPDefinition>,CPDefinitionOptionRelService,CPInstanceService,CPInstanceHelper
     [echo]   reason: actual and formal argument lists differ in length
     [echo] Note: Some input files use or override a deprecated API.
     [echo] Note: Recompile with -Xlint:deprecation for details.
     [echo] 6 errors
     [echo] 
     [echo] FAILURE: Build failed with an exception.
     [echo] 
     [echo] * What went wrong:
     [echo] Execution failed for task ':apps:commerce:commerce-product-definitions-web:compileJava'.
     [echo] > Compilation failed; see the compiler error output for details.
     [echo] 
     [echo] * Try:
     [echo] Run with --info or --debug option to get more log output.
     [echo] 
     [echo] * Exception is:
     [echo] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':apps:commerce:commerce-product-definitions-web:compileJava'.
```

**Comment by alessiorendina** [comment]
@pyoo47 thanks :)

**Comment by alessiorendina** [comment]
ci:retest

**Comment by ethanbustad** [comment]
ci:test

**Comment by ethanbustad** [comment]
Some more compileJSP issues:
```
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/view_005fspecification_005foptions_jsp.java:367: error: cannot find symbol
     [echo]       _jspx_th_c_if_0.setTest( cpOptionDisplayContext.hasPermission(CPActionKeys.MANAGE_CATALOG) );
     [echo]                                ^
     [echo]   symbol:   variable cpOptionDisplayContext
     [echo]   location: class view_005fspecification_005foptions_jsp
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/view_005foption_005fcategories_jsp.java:361: error: cannot find symbol
     [echo]       _jspx_th_c_if_0.setTest( cpOptionDisplayContext.hasPermission(CPActionKeys.MANAGE_CATALOG) );
     [echo]                                ^
     [echo]   symbol:   variable cpOptionDisplayContext
     [echo]   location: class view_005foption_005fcategories_jsp
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/edit_005fspecification_005foption_jsp.java:363: error: cannot find symbol
     [echo] if (hasManageCPSpecificationOptionsPermission) {
     [echo]     ^
     [echo]   symbol:   variable hasManageCPSpecificationOptionsPermission
     [echo]   location: class edit_005fspecification_005foption_jsp
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/edit_005fspecification_005foption_jsp.java:363: error: illegal start of type
     [echo] if (hasManageCPSpecificationOptionsPermission) {
     [echo]    ^
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/edit_005fspecification_005foption_jsp.java:367: error: cannot find symbol
     [echo] if (hasManageCPOptionCategoriesPermission) {
     [echo]     ^
     [echo]   symbol:   variable hasManageCPOptionCategoriesPermission
     [echo]   location: class edit_005fspecification_005foption_jsp
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/edit_005fspecification_005foption_jsp.java:367: error: illegal start of type
     [echo] if (hasManageCPOptionCategoriesPermission) {
     [echo]    ^
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/edit_005foption_005fcategory_jsp.java:363: error: cannot find symbol
     [echo] if (hasManageCPSpecificationOptionsPermission) {
     [echo]     ^
     [echo]   symbol:   variable hasManageCPSpecificationOptionsPermission
     [echo]   location: class edit_005foption_005fcategory_jsp
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/edit_005foption_005fcategory_jsp.java:363: error: illegal start of type
     [echo] if (hasManageCPSpecificationOptionsPermission) {
     [echo]    ^
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/edit_005foption_005fcategory_jsp.java:367: error: cannot find symbol
     [echo] if (hasManageCPOptionCategoriesPermission) {
     [echo]     ^
     [echo]   symbol:   variable hasManageCPOptionCategoriesPermission
     [echo]   location: class edit_005foption_005fcategory_jsp
     [echo] /opt/dev/projects/github/liferay-portal-7.1.x/modules/apps/commerce/commerce-product-options-web/build/jspc/org/apache/jsp/edit_005foption_005fcategory_jsp.java:367: error: illegal start of type
     [echo] if (hasManageCPOptionCategoriesPermission) {
     [echo]    ^```

**Comment by alessiorendina** [comment]
@ethanbustad Did it, hope this will fix everything! Thanks!

**Comment by alessiorendina** [comment]
ci:retest

**Comment by alessiorendina** [comment]
ci:retest

**Pull request merged**